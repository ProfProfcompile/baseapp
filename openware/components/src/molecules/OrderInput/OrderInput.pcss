.cr-order-input {
  display: flex;
  justify-content: space-between;

  &__crypto-icon {
    background-color: var(--order-input-input-bg);
    border: 1px solid var(--order-input-border-color);
    border-right: none;
    display: inline-block;
    padding: var(--gap);
    width: 50%;

    span {
      background-color: inherit;
    }
  }

  &__input {
    background-color: var(--order-input-input-bg);
    border: 1px solid var(--order-input-border-color);
    color: var(--order-input-input-color);
    display: inline-block;
    text-align: right;
    width: 50%;

    .cr-input {
      border: none;
      padding: var(--gap);
      width: auto;
    }
  }
}
