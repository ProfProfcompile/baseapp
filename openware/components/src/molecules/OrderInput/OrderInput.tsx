import classnames from 'classnames';
import * as React from 'react';
import { CryptoIcon, Input } from '../../atoms';

interface OrderInputProps {
    /**
     * Additional class name for styling. By default element receives `cr-input-block` class
     * @default empty
     */
    className?: string;
    /**
     * Code of cryptocurrency
     * @default empty
     */
    currency: string;
    /**
     * Value of Input component
     */
    value: string | number;
    /**
     * Function for getting value of input
     * @default empty
     */
    handleChangeValue: (text: string) => void;
}

/**
 * Input with cryptocurrency icon and label.
 */
const OrderInput: React.FunctionComponent<OrderInputProps> = (props: OrderInputProps) => {
    const { currency, className, value, handleChangeValue } = props;
    const cx = classnames('cr-order-input', className);
    const currencyCode = `${currency.toUpperCase()}-alt`;
    return (
        <div className={cx}>
            <div className="cr-order-input__crypto-icon">
                <CryptoIcon code={currencyCode}>{currency.toUpperCase()}</CryptoIcon>
            </div>
            <div className="cr-order-input__input">
                <Input
                    className="cr-order-input__input"
                    type="text"
                    placeholder="0"
                    value={value}
                    onChangeValue={handleChangeValue}
                />
            </div>
        </div>
    );
};

export {
    OrderInput,
    OrderInputProps,
};
