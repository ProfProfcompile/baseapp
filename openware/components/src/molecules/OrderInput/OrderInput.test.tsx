import { shallow } from 'enzyme';
import * as React from 'react';
import { OrderInput, OrderInputProps } from './OrderInput';

const defaultProps: OrderInputProps = {
    currency: 'eth',
    handleChangeValue: () => undefined,
    value: '',
};

const setup = (props: Partial<OrderInputProps> = {}) =>
    shallow(<OrderInput {...{ ...defaultProps, ...props }} />);

describe('InputBlock', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        const wrapper = setup();
        expect(wrapper.hasClass('cr-order-input')).toBeTruthy();
    });

    it('should pass along supplied className', () => {
        const className = 'some-class';
        const wrapper = setup({ className });
        expect(wrapper.hasClass(className)).toBeTruthy();
    });

    it('should set correct currency code to the child components', () => {
        const wrapper = setup();
        const { children } = wrapper.find('CryptoIcon').props();
        expect(children).toContain('ETH');
    });
});
