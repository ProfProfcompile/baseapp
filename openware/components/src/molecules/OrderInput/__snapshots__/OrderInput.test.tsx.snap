// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBlock should render 1`] = `
<div
  className="cr-order-input"
>
  <div
    className="cr-order-input__crypto-icon"
  >
    <CryptoIcon
      code="ETH-alt"
    >
      ETH
    </CryptoIcon>
  </div>
  <div
    className="cr-order-input__input"
  >
    <Input
      className="cr-order-input__input"
      onChangeValue={[Function]}
      placeholder="0"
      type="text"
      value=""
    />
  </div>
</div>
`;
